src/lib/coda_base/transaction_union_payload.ml

```diff
@@ -231,3 +231,14 @@ let supply_increase (payload : payload) =
       payload.body.amount
   | Payment | Stake_delegation | Create_account | Fee_transfer ->
       Amount.zero
+
+let next_available_token (payload : payload) tid =
+  match payload.body.tag with
+  | Payment | Stake_delegation | Coinbase | Fee_transfer ->
+      tid
+  | Create_account when Token_id.(equal invalid) payload.body.token_id ->
```

Comment:
is a tid invalid only in the case it hasn't yet been allocated? Is there ever a de-allocation?

Reply:
Yea, `tid = invalid` signals to the transaction snark that it should create a token as well as creating a new account. At the moment, it isn't (or at least shouldn't be) used for anything else.